docs/java-getting-started.md

```diff
@@ -115,44 +113,104 @@ To create a local copy of a sample app that you can deploy to Heroku, execute th
 :::-- $ git merge origin/main
 ```
 
-You now have a functioning Git repository that contains a simple Java application. The application includes a `pom.xml` file, which is used by Java's dependency manager, Maven.
+This functioning Git repository contains a simple Java application. The application includes a `Procfile`, a special plaintext file used by Heroku apps. You explicitly declare the processes and commands used to start your app in this file.
+
+The `Procfile` in the example app source code looks like this:
+
+```yaml
+:::-> $ cat Procfile
+```
+
+This file declares a single process type, `web`, and the command needed to run it. The name `web` is important. It declares that this process type attaches to Heroku's [HTTP routing](http-routing) stack, and is able to receive web traffic.
 
-<h2 data-next-message="I have deployed my app on Heroku">Deploy the app</h2>
+Procfiles can contain additional process types. For example, you can declare a background worker that processes items off a queue. This tutorial doesn't cover other processes but you can refer to [The Procfile](procfile) and [The Process Model](process-model) for more info.
 
-In this step you'll deploy the sample app to Heroku.
+The example app also includes a `pom.xml` file, which is used by Java's dependency manager, Maven. The next step covers how to use this file to declare dependencies. 
+
+<h2 data-next-message="I inspected the app dependencies">Declare App Dependencies</h2>
+
+Heroku automatically identifies an app as a Java app if it contains a `pom.xml` file in the root directory. When a Java app is detected, Heroku adds the official Java buildpack to your app, which installs the dependencies for your application. You can create a `pom.xml` file for your own apps with the `mvn archetype:generate` command.
+
+The example app you deployed already has a `pom.xml` ([see it here](https://github.com/heroku/java-getting-started/blob/main/pom.xml)). Here's an excerpt:
+
+```xml
+:::-> $ sed -n '27,35p' pom.xml
+```
 
-First, create an app on Heroku, which prepares Heroku to receive your source code:
+When deploying an app, Heroku reads this file and installs the dependencies by running `mvn clean install`. Take a look at the dependencies listed in your `pom.xml`. 
+
+Another file, `system.properties`, indicates the version of Java to use. The contents of this optional file look like:
+
+```
+:::-> $ cat system.properties
+```
+
+Heroku supports many [different versions](java-support#supported-java-versions). You can push your own apps using a different version of Java.
+
+<h2 data-next-message="I deployed my app on Heroku">Deploy the App</h2>
+
+Create an app on Heroku to prepare it to receive your source code for deployment:
```

Comment:
I'm trying to get rid of the "In this step" sentences at the beginning of each step as they just repeat the step/section heading. We also want to avoid using future tense as much as possible to fit our writing guidelines. How about just "Create an app on Heroku to prepare it to receive your source code:"?

Reply:
It's okay I think we can just keep what you have currently then. :)